commands/herolevel.js

```diff
@@ -1,345 +1,158 @@
-                    finalArr.push(levelUpRound - 1);
-                }
-            }
-            const embed = new Discord.MessageEmbed()
-                .setTitle(heroname)
-                .setDescription(
-                    'This shows which round the hero will reach which level'
-                )
-                .addField('level 1', `r${finalArr[0]}`, true)
-                .addField('level 2', `r${finalArr[1]}`, true)
-                .addField('level 3', `r${finalArr[2]}`, true)
-                .addField('level 4', `r${finalArr[3]}`, true)
-                .addField('level 5', `r${finalArr[4]}`, true)
-                .addField('level 6', `r${finalArr[5]}`, true)
-                .addField('level 7', `r${finalArr[6]}`, true)
-                .addField('level 8', `r${finalArr[7]}`, true)
-                .addField('level 9', `r${finalArr[8]}`, true)
-                .addField('level 10', `r${finalArr[9]}`, true)
-                .addField('level 11', `r${finalArr[10]}`, true)
-                .addField('level 12', `r${finalArr[11]}`, true)
-                .addField('level 13', `r${finalArr[12]}`, true)
-                .addField('level 14', `r${finalArr[13]}`, true)
-                .addField('level 15', `r${finalArr[14]}`, true)
-                .addField('level 16', `r${finalArr[15]}`, true)
-                .addField('level 17', `r${finalArr[16]}`, true)
-                .addField('level 18', `r${finalArr[17]}`, true)
-                .addField('level 19', `r${finalArr[18]}`, true)
-                .addField('level 20', `r${finalArr[19]}`, true)
-                .setColor(cyber);
-            return embed;
+const AnyOrderParser = require('../parser/any-order-parser.js');
+const OptionalParser = require('../parser/optional-parser.js');
+
+const HeroParser = require('../parser/hero-parser.js');
+const RoundParser = require('../parser/round-parser');
+const MapDifficultyParser = require('../parser/map-difficulty-parser.js');
+
+const ReactionChain = require('../reactor/reaction_chain')
+const EmojiReactor = require('../reactor/emoji_reactor')
+const SingleTextParser = require('../reactor/single_text_parser')
+
+function execute(message, args) {
+    if (args.length == 1 && args[0] == 'help') {
+        return message.channel.send('Type `q!herolevel` and follow the instructions');
+    }
+
+    const parsed = CommandParser.parse(
+        args,
+        
+        // Make any of the available arguments optional to add in any order in the command args
+        // Arguments that aren't entered will be gathered through the react-loop
+        new AnyOrderParser(
+            new OptionalParser(
+            new HeroParser()
+            ),
+            new OptionalParser(
+                new RoundParser('ALL')
+            ),
+            new OptionalParser(
+                new MapDifficultyParser()
+            )
+        )
+    );
+
+    if (parsed.hasErrors()) {
+        return errorMessage(message, parsed.parsingErrors)
+    }
+
```

Comment:
See below how simplified it becomes to collect reactions and individual pieces of data using the `ReactChain` module!